presto-main/src/main/java/com/facebook/presto/operator/MoreByteArrays.java

```diff
@@ -0,0 +1,100 @@
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.api.Experimental;
+import sun.misc.Unsafe;
+
+import java.lang.reflect.Field;
+
+import static io.airlift.slice.SizeOf.SIZE_OF_BYTE;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static sun.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+import static sun.misc.Unsafe.ARRAY_INT_INDEX_SCALE;
+
+@Experimental
+public class MoreByteArrays
+{
+    private static final Unsafe unsafe;
+
+    static {
+        try {
+            // fetch theUnsafe object
+            Field field = Unsafe.class.getDeclaredField("theUnsafe");
+            field.setAccessible(true);
+            unsafe = (Unsafe) field.get(null);
+            if (unsafe == null) {
+                throw new RuntimeException("Unsafe access not available");
+            }
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static byte getByte(byte[] bytes, int index)
+    {
+        checkValidRange(index, SIZE_OF_BYTE, bytes.length);
+        return unsafe.getByte(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET);
+    }
+
+    public static int fill(byte[] bytes, int index, int length, byte value)
+    {
+        requireNonNull(bytes, "bytes is null");
+        checkValidRange(index, length, bytes.length);
+
+        unsafe.setMemory(bytes, index + ARRAY_BYTE_BASE_OFFSET, length, value);
+        return index + length;
+    }
+
+    public static int setBytes(byte[] bytes, int index, byte[] values, int offset, int length)
+    {
+        requireNonNull(bytes, "bytes is null");
+        requireNonNull(values, "values is null");
+
+        checkValidRange(index, length, bytes.length);
+        checkValidRange(offset, length, values.length);
+
+        // The performance of one copy and two copies (one big chunk at 8 bytes boundary + rest) are about the same.
+        unsafe.copyMemory(values, (long) offset + ARRAY_BYTE_BASE_OFFSET, bytes, (long) index + ARRAY_BYTE_BASE_OFFSET, length);
+        return index + length;
+    }
+
+    public static int setInts(byte[] bytes, int index, int[] values, int offset, int length)
+    {
+        requireNonNull(bytes, "bytes is null");
+        requireNonNull(values, "values is null");
+
+        checkValidRange(index, length * ARRAY_INT_INDEX_SCALE, bytes.length);
+        checkValidRange(offset, length, values.length);
+
+        for (int i = offset; i < offset + length; i++) {
+            unsafe.putInt(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET, values[i]);
+            index += ARRAY_INT_INDEX_SCALE;
+        }
+        return index;
+    }
+
+    private static void checkValidRange(int start, int length, int size)
+    {
+        if (start < 0 || length < 0 || start + length > size) {
+            throw new IndexOutOfBoundsException(format("Invalid start %s and length %s with array size %s", start, length, size));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: size (declared at line 91), start (declared at line 91), length (declared at line 91). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 94. Consider formatting this data with &#39;%d&#39; instead.